packages/core/src/routes/swagger.ts

```diff
@@ -1,51 +1,83 @@
+  const guard = stack.find((function_): function_ is WithGuardConfig<IMiddleware> =>
+    isGuardMiddleware(function_)
+  );
+  const body = guard?.config.body;
+
+  return {
+    tags: [toTitle(path.split('/')[1] ?? 'General')],
+    requestBody: body && {
+      required: true,
+      content: {
+        'application/json': {
+          schema: zodTypeToSwagger(body),
+        },
+      },
+    },
+    responses: {
+      '200': {
+        description: 'OK',
+      },
+    },
+  };
+};
+
+export default function swaggerRoutes<T extends AnonymousRouter, R extends Router<unknown, any>>(
+  router: T,
+  allRouters: R[]
+) {
   router.get('/swagger.json', async (ctx, next) => {
-    const routes = ctx.router.stack.map(({ path, stack, methods }) => {
-      const guard = stack.find((function_): function_ is WithGuardConfig<IMiddleware> =>
-        isGuardMiddleware(function_)
-      );
+    const routes = allRouters.flatMap<RouteObject>((router) =>
+      router.stack.flatMap<RouteObject>(({ path, stack, methods }) =>
+        methods
+          // There is no need to show the HEAD method.
+          .filter((method) => method !== 'HEAD')
+          .map((method) => ({
+            path: `/api${path}`,
+            method: method.toLowerCase() as HttpMethod,
+            operation: buildOperation(stack, path),
+          }))
+      )
+    );
 
-      return { path, methods, guard };
-    });
+    // Group routes by path
+    // eslint-disable-next-line unicorn/prefer-object-from-entries
+    const paths = routes.reduce<PathsObject>(
+      (pathsObject, { path, method, operation }: RouteObject) => {
+        const methodObject = pathsObject[path];
 
-    const paths = Object.fromEntries(
-      routes.map<[string, OpenAPIV3.PathItemObject]>(({ path, methods, guard }) => {
-        const body = guard?.config.body;
+        /* eslint-disable @silverhand/fp/no-mutation */
+        if (methodObject) {
+          methodObject[method] = operation;
+        } else {
+          pathsObject[path] = { [method]: operation };
+        }
+        /* eslint-enable @silverhand/fp/no-mutation */
 
-        return [
-          `/api${path}`,
-          Object.fromEntries(
-            methods.map<[string, OpenAPIV3.OperationObject]>((method) => [
-              method.toLowerCase(),
-              {
-                tags: [toTitle(path.split('/')[1] ?? 'General')],
-                requestBody: body && {
-                  required: true,
-                  content: {
-                    'application/json': {
-                      schema: zodTypeToSwagger(body),
-                    },
-                  },
-                },
-                responses: {
-                  '200': {
-                    description: 'OK',
-                  },
-                },
-              },
-            ])
-          ),
-        ];
-      })
+        return pathsObject;
+      },
+      Object.create(null)
```

Comment:
nit: what's the difference between {} and Object.create(null)?

Reply:
<img width="670" alt="image" src="https://user-images.githubusercontent.com/10594507/171995469-fdb1fb98-03cf-4063-ae38-cd3f5dbc851e.png">

`Object.create()` will not be used after refactoring according to other comments

Reply:
I mean under your case, not following why have to be `Object.create(null)`

Reply:
To be more clear. Whenever a js object is initially assigned it contains the prototype properties by default, is there any specific reasons you want to remove it.

Reply:
i create it with no properties from Object.prototype on purpose, so we can use it like a simple map/dict. under this case, it's enough to use.